paderbox/utils/random_utils.py

```diff
@@ -23,19 +23,47 @@
 ]
 
 
-def str_to_random_state(string):
+def str_to_seed(string: str, bits: int = 32) -> int:
     """
-    This functions outputs a consistent random state dependent on
-    an input string.
-    
-    >>> print(str_to_random_state(''))
-    RandomState(MT19937)
+    This functions outputs a consistent seed in the range (0, 2**bits - 1)
+    dependent on an input string.
+
+    >>> print(str_to_seed(''))
+    2018687061
     """
     import hashlib
     hash_value = hashlib.sha256(string.encode("utf-8"))
     hash_value = int(hash_value.hexdigest(), 16)
-    hash_value = hash_value % 2 ** 32
-    return np.random.RandomState(hash_value)
+    hash_value = hash_value % 2 ** bits
+    return hash_value
+
+
+def str_to_random_state(
+        string: str,
+        seed_bits: int = 32
+) -> np.random.RandomState:
+    """
+    This functions outputs a consistent random state (`np.random.RandomState`)
+    dependent on an input string.
+    
+    >>> print(str_to_random_state(''))
+    RandomState(MT19937)
+    """
+    return np.random.RandomState(str_to_seed(string, bits=seed_bits))
+
+
+def str_to_random_generator(
+        string: str,
+        seed_bits: int = 128
+) -> np.random.Generator:
```

Comment:
Do you know, when `np.random.Generator` was introduced?
How about changiing this to `'np.random.Generator'` (i.e. a `str`)?
It would then work for old configurations of python (i.e. less than 3.7, see PEP 563 -- Postponed Evaluation of Annotations) and numpy.

Reply:
Done